src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -456,116 +484,133 @@ private async Task AcceptStreamsAsync()
+                    case (byte)Http3StreamType.Push:
+                        // We don't support push streams.
+                        // Because no maximum push stream ID was negotiated via a MAX_PUSH_ID frame, server should not have sent this. Abort the connection with H3_ID_ERROR.
+                        throw HttpProtocolException.CreateHttp3ConnectionException(Http3ErrorCode.IdError);
+                    case (long)Http3StreamType.WebTransportBidirectional:
+                    case (long)Http3StreamType.WebTransportUnidirectional:
+                        if (EnableWebTransport)
+                        {
+                            VariableLengthIntegerHelper.TryRead(buffer.ActiveSpan.Slice(bytesRead), out long sessionId, out bytesRead);
+                            quicStream = null;
+                            WebtransportManager!.AcceptServerStream(stream, sessionId);
                             return;
-                        case (byte)Http3StreamType.QPackDecoder:
-                            if (Interlocked.Exchange(ref _haveServerQpackDecodeStream, 1) != 0)
-                            {
-                                // A second QPack decode stream has been received.
-                                throw HttpProtocolException.CreateHttp3ConnectionException(Http3ErrorCode.StreamCreationError);
-                            }
+                        }
+                        stream.Abort(QuicAbortDirection.Read, (long)Http3ErrorCode.StreamCreationError);
+                        return;
+                    default:
+                        // Server initiated bidirectional streams are either push streams or extensions, and we support neither.
+                        if (stream.CanWrite)
+                        {
+                            throw HttpProtocolException.CreateHttp3ConnectionException(Http3ErrorCode.StreamCreationError);
+                        }
+                        // Unknown stream type. Per spec, these must be ignored and aborted but not be considered a connection-level error.
 
-                            // The stream must not be closed, but we aren't using QPACK right now -- ignore.
-                            buffer.Dispose();
-                            await stream.CopyToAsync(Stream.Null).ConfigureAwait(false);
-                            return;
-                        case (byte)Http3StreamType.QPackEncoder:
-                            if (Interlocked.Exchange(ref _haveServerQpackEncodeStream, 1) != 0)
-                            {
-                                // A second QPack encode stream has been received.
-                                throw HttpProtocolException.CreateHttp3ConnectionException(Http3ErrorCode.StreamCreationError);
-                            }
+                        if (NetEventSource.Log.IsEnabled())
+                        {
+                            // Read the rest of the integer, which might be more than 1 byte, so we can log it.
```

Comment:
Do you need this logic now?

Reply:
No, I will just log the stream type since we read it all now

Reply:
No I do not, left only the log